### ISSUE TITLE
test-run doesn't kill hung tests in the single process mode

### ISSUE THREAD

**Issue by locker** [tarantool/test-run#106]
Reprodcued by [Travis CI](https://travis-ci.org/tarantool/tarantool/jobs/411186189):
```
vinyl/constraint.test.lua                                       [ pass ]
vinyl/ddl.test.lua                                              
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated
```

**Comment by locker** [comment]
It would also be great if test-run printed the test output (both for tap and diff tests) after killing it on timeout. Should I file another issue?

**Comment by locker** [comment]
A note: test-run does kill a hung test if called without options (e.g. `./test-run box/cfg.test.lua`), but does not if called in the single process mode (i.e. `./test-run -j -1 box/cfg.test.lua`).

**Comment by Totktonada** [comment]
Maybe it worth to just use `-j 1` in Travis CI instead of `-j -1`. I don’t know how this quasi-parallel mode can increase tests flakiness.

@sergw Maybe you want to participare in this decision.

Open question: can we estimate level of tests flakiness increase after the switch with current level of flakiness?

**Comment by sergw** [comment]
It will be good to change mode to `-j 1`:
- one step out from legacy consistent `test-run`
- behaviour and output same as in parallel
- easy to contribute & bug fix (one place to change)


> can we estimate level of tests flakiness

I gueess it will be the same, cause as I see, flakiness depends on tests not the `test-run`.

**Comment by Totktonada** [comment]
I propose to change tarantool CI (`.travis.mk` and `rpm/tarantool.spec` files) w/o actual changes of test-run and estimate the effect. Then, maybe, remove the single process mode, because it going to be unmaintained.

**Comment by Totktonada** [comment]
@sergw Can be considered as worked around and be closed as won’t fix?
